ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -229,6 +246,13 @@ func (r *RayClusterReconciler) rayClusterReconcile(ctx context.Context, instance
 		return ctrl.Result{}, nil
 	}
 
+	if err := validateRayClusterSpec(instance); err != nil {
+		logger.Error(err, "The RayCluster spec is invalid")
+		r.Recorder.Eventf(instance, corev1.EventTypeWarning, string(utils.InvalidRayClusterSpec),
+			"The RayCluster spec is invalid %s/%s: %v", instance.Namespace, instance.Name, err)
+		return ctrl.Result{}, err
```

Comment:
```suggestion
		return ctrl.Result{}, nil
```
I guess returning an err here still requeues the invalid CR. Could you double check on that?